### DISCUSSION TITLE
There is nucleation signal, but the corresponding reserved order parameter can not be observed

### DISCUSSION THREAD

**Discussion by YXY101676** [idaholab/moose#27935] [Q&A Modules: Phase field]
Here is my input file：
![image](https://github.com/idaholab/moose/assets/154942901/84e7f815-5a19-43f5-b40a-c3e1b6d76f9b)
![image](https://github.com/idaholab/moose/assets/154942901/1ee4018b-f303-4eaf-83c0-12347dd67855)
![image](https://github.com/idaholab/moose/assets/154942901/15951100-7ea4-405e-be51-5c1b820ae742)
The nucleation signal can be seen, but the corresponding order parameter gr9 is 0
![image](https://github.com/idaholab/moose/assets/154942901/5ac7fb51-c83e-4a2f-81ca-7a92fa50636f)
![image](https://github.com/idaholab/moose/assets/154942901/74cf0443-845e-4cba-b176-2d4772054b2d)

**Comment by YXY101676** [comment]
Does this mean that the preserving order parameter of the newly inserted nucleation is GR0？
![image](https://github.com/idaholab/moose/assets/154942901/08f778be-ad30-4c2f-985b-acde8996f490)

**Reply by GiudGiud** [reply]
@permcody on grain nucleation

**Reply by permcody** [reply]
If I recall correctly, once you see that message, the GrainTracker will "move" your grain from the reserved order parameter to one of the "normal" order parameters to continue evolving it. Try looking at the reserved order parameter (in your case gr9) one time step earlier, unless your insertion is happening in _this_ step. In any case, visualizing the staging area may not be very interesting. There should be a message about which OP the new grain got remapped to right after that message.
I assume you found the page on Nucleation here: https://mooseframework.inl.gov/modules/phase_field/Nucleation/DiscreteNucleation.html

There are also a few test cases checked in for nucleation that are very small and run quickly. You might run those and observe the behavior along with the screen output. It will help you understand a bit more of what's going on in your simulation.

**Reply by YXY101676** [reply]
https://mooseframework.inl.gov/modules/phase_field/Nucleation/DiscreteNucleation.html
I learned about nucleation on the above page, but there were no examples of polycrystalline nucleation, so I couldn't get more specific guidance. 
And in my case, at Step 27, there's a nucleation signal:
![image](https://github.com/idaholab/moose/assets/154942901/7e154485-1b50-4be7-8f0f-ece5ad6a5da1)
But I couldn't find the output of gr7 in the visualizations of steps 26 and 27, and there was no signal like a rema.
There is no output from GR9 until step 30, and there is always evolution：
![image](https://github.com/idaholab/moose/assets/154942901/8f84cb8a-336c-463d-b144-d6d4f03b5162)
![image](https://github.com/idaholab/moose/assets/154942901/46799ef4-e5cd-4fdc-a20f-c361b7cc41bc)

**Comment by permcody** [comment]
Oh… I don’t like seeing those “split grain detected” messages. Something
isn’t right with what’s being detected.

Either I or somebody else is going to have to dig into the code, with this
input. I hate to say this, but you might try a different random seed to see
if you can get past this. It might be the case that I just disable split
grain detection in a simulation where we shouldn’t expect that (most if not
all Voroni-based initial conditions).

**Reply by YXY101676** [reply]
So I should change the initial grain structure to avoid the crack problem? What does this GR9 stand for? Aren't they new nucleated grains

**Reply by YXY101676** [reply]
Does anyone use moose to complete the nucleation and grain growth process of polycrystalline structure? This is very important to me

**Reply by permcody** [reply]
While definitely not a solution, changing the initial grain structure may get you past the immediate problem so it’s worth a try.  GR9 is simply the name of one of your order parameter fields. In your input file your input file, you requested 10 order parameters with a prefix of “gr” so they are gr0-gr9. The highest numbered order parameter will be the reserved field for nucleation. So to answer your last question. It’s not a nucleated grain, it’s the field that holds your grains as the are starting to nucleate until they are large enough to recognize and transfer.

We have done polycrystalline simulations with grain nucleation. @dschwen, do we have an sample input files for that kind of sim?

**Reply by YXY101676** [reply]
But the holdtime I set it to 0, it shouldn't actually be here this long

**Reply by YXY101676** [reply]
To what extent will be mapped by graintracker, is it set by the reserve parameter
![Uploading image.png…]()
